tests/lib/prepare.sh

```diff
@@ -518,3 +529,266 @@ prepare_all_snap() {
+
+            # create /var/lib/extrausers/$f
+            # append ubuntu, test user for the testing
+            grep "^test:" /etc/$f >> /mnt/system-data/var/lib/extrausers/"$f"
+            grep "^ubuntu:" /etc/$f >> /mnt/system-data/var/lib/extrausers/"$f"
+            # check test was copied
+            MATCH "^test:" </mnt/system-data/var/lib/extrausers/"$f"
+            MATCH "^ubuntu:" </mnt/system-data/var/lib/extrausers/"$f"
+        done
+
+        # ensure spread -reuse works in the core image as well
+        if [ -e /.spread.yaml ]; then
+            cp -av /.spread.yaml /mnt/system-data
+        fi
+
+        # using symbolic names requires test:test have the same ids
+        # inside and outside which is a pain (see 12345 above), but
+        # using the ids directly is the wrong kind of fragile
+        chown --verbose test:test /mnt/user-data/test
+
+        # we do what sync-dirs is normally doing on boot, but because
+        # we have subdirs/files in /etc/systemd/system (created below)
+        # the writeable-path sync-boot won't work
+        mkdir -p /mnt/system-data/etc/systemd
+        (cd /tmp ; unsquashfs -v  /var/lib/snapd/snaps/core18_*.snap etc/systemd/system)
+        cp -avr /tmp/squashfs-root/etc/systemd/system /mnt/system-data/etc/systemd/
+
+
+        umount /mnt
+        kpartx -d  "$IMAGE_HOME/$IMAGE"
+
+        # the reflash magic
+        # FIXME: ideally in initrd, but this is good enough for now
+        cat > "$IMAGE_HOME/reflash.sh" << EOF
+#!/bin/sh -ex
+mount -t tmpfs none /tmp
+cp /bin/busybox /tmp
+cp $IMAGE_HOME/$IMAGE /tmp
+sync
+# blow away everything
+/tmp/busybox dd if=/tmp/$IMAGE of=/dev/sda bs=4M
+# and reboot
+/tmp/busybox sync
+/tmp/busybox echo b > /proc/sysrq-trigger
+EOF
+        chmod +x "$IMAGE_HOME/reflash.sh"
+
+        # extract ROOT from /proc/cmdline
+        ROOT=$(sed -e 's/^.*root=//' -e 's/ .*$//' /proc/cmdline)
+        cat >/boot/grub/grub.cfg <<EOF
+set default=0
+set timeout=2
+menuentry 'flash-all-snaps' {
+linux /vmlinuz root=$ROOT ro init=$IMAGE_HOME/reflash.sh console=ttyS0
+initrd /initrd.img
+}
+EOF
+}
+
+prepare_core18() {
+    # we are still a "classic" image, prepare the surgery
+    if [ -e /var/lib/dpkg/status ]; then
+        setup_reflash_magic_core18
+        REBOOT
+    fi
+
+    # verify after the first reboot that we are now in core18 world
+    if [ "$SPREAD_REBOOT" = 1 ]; then
+        echo "Ensure we are now in an all-snap world"
+        if [ -e /var/lib/dpkg/status ]; then
+            echo "Rebooting into all-snap system did not work"
+            exit 1
+        fi
+    fi
+
+    echo "Wait for firstboot change to be ready"
+    while ! snap changes | grep "Done"; do
+        snap changes || true
+        snap change 1 || true
+        sleep 1
+    done
+
+    echo "Ensure fundamental snaps are still present"
+    # shellcheck source=tests/lib/names.sh
+    . "$TESTSLIB/names.sh"
+    for name in "$gadget_name" "$kernel_name" core18 snapd; do
```

Comment:
you could use $core_name here instead of core18, right?

Reply:
Nice one, yes, fixed!